File: Emulator/src/main/content/global/skill/summoning/familiar/npc/FireTitanNPC.java
package content.global.skill.summoning.familiar.npc;

import content.global.skill.summoning.familiar.Familiar;
import core.game.node.entity.combat.equipment.WeaponInterface;
import core.game.node.entity.player.Player;
import core.plugin.Initializable;

/**
 * The type Fire titan npc.
 */
@Initializable
public class FireTitanNPC extends ElementalTitanNPC {

    /**
     * Instantiates a new Fire titan npc.
     */
    public FireTitanNPC() {
        this(null, 7355);
    }

    /**
     * Instantiates a new Fire titan npc.
     *
     * @param owner the owner
     * @param id    the id
     */
    public FireTitanNPC(Player owner, int id) {
        super(owner, id, 6200, 12802, 20, WeaponInterface.STYLE_CAST);
    }

    @Override
    public Familiar construct(Player owner, int id) {
        return new FireTitanNPC(owner, id);
    }

    @Override
    public int[] getIds() {
        return new int[]{7355, 7356};
    }

}

File: Emulator/src/main/content/global/skill/summoning/familiar/npc/FireTitanNPC.kt
package content.global.skill.summoning.familiar.npc

import content.global.skill.summoning.familiar.Familiar
import core.game.node.entity.combat.equipment.WeaponInterface
import core.game.node.entity.player.Player
import core.plugin.Initializable
import shared.consts.Items
import shared.consts.NPCs

@Initializable
class FireTitanNPC @JvmOverloads constructor(owner: Player? = null, id: Int = NPCs.FIRE_TITAN_7355) :
    ElementalTitanNPC(owner, id, 6200, Items.FIRE_TITAN_POUCH_12802, 20, WeaponInterface.STYLE_CAST) {
    override fun construct(owner: Player, id: Int): Familiar {
        return FireTitanNPC(owner, id)
    }

    override fun getIds(): IntArray {
        return intArrayOf(NPCs.FIRE_TITAN_7355, NPCs.FIRE_TITAN_7356)
    }
}
